File: python/isbn-verifier/isbn_verifier.py
def verify(isbn):
    isbn = isbn.replace('-', '')

    if len(isbn) != 10:
        return False

    try:
        isbn = [10 if n == 'X' else int(n) for n in list(isbn)]

        if 10 in isbn and isbn[-1] != 10:
            return False

        check_numbers = list(zip(isbn, list(range(10, 0, -1))))
        check = [n[0] * n[1] for n in check_numbers]
        return (sum(check) % 11) == 0
    except ValueError:
        return False

File: elixir/isbn-verifier/isbn_verifier.exs
defmodule ISBNVerifier do
  @doc """
    Checks if a string is a valid ISBN-10 identifier

    ## Examples

      iex> ISBNVerifier.isbn?("3-598-21507-X")
      true

      iex> ISBNVerifier.isbn?("3-598-2K507-0")
      false

  """
  @spec isbn?(String.t()) :: boolean
  def isbn?(isbn) do
    isbn = String.replace(isbn, "-", "")
    cond do
      String.match?(isbn, ~r/[^0-9X]/) ->
        false
      true ->
        isbn
        |> String.graphemes()
        |> Enum.map(fn c ->
          case c do
            "X" -> 10
            _ -> String.to_integer(c)
          end
        end)
        |> Enum.zip(Enum.to_list(10..1))
        |> Enum.map(fn {a, b} -> a * b end)
        |> Enum.sum()
        |> rem(11)
        |> Kernel.==(0)
    end
  end
end
